app/src/stores/fields.ts

```diff
@@ -72,6 +72,20 @@ export const useFieldsStore = defineStore({
 		parseField(field: FieldRaw): Field {
 			let name = formatTitle(field.field);
 
+			const localesToReset = !isNil(field.meta?.translations)
```

Comment:
![image](https://user-images.githubusercontent.com/26413686/197445376-60eca25e-e3f8-430f-b853-caa9bda934e8.png)

Adding a check for `field.meta!.translations?.length > 0` similar to the block after seems to fix the crashing bug. It seems that the typing is incorrect as the `field.meta.translations` could have a non-array value. 🤔 

![image](https://user-images.githubusercontent.com/26413686/197446166-527ad67b-2b7c-45a6-976d-0f230ae1e6ea.png)

Reply:
Have you restarted the API? This seems like the issue with `api/src/database/system-data/fields/settings.yaml` not using array syntax (and seems very outdated), which was also fixed in this PR.

Technically the length check does also fix this, but I'm also not sure should we presume that it can _not_ be an array, aside from this outdated directus_settings file 🤔

Reply:
Ooo yes, restarting resolves the issue. 👍 I think it should be fine then, since the cause of this typing issue is due to the obsolete yaml config. Thanks!

Reply:
Just to add on a bit, partially didn't added the length check as sort of an attempt to "make them more similar" since collections doesn't have it:

https://github.com/directus/directus/blob/e4364df25bb793a1c0c77a852d6354b9eb090a1d/app/src/stores/collections.ts#L56

but in hindsight, maybe lodash `isEmpty` is more concise for null and empty array 🧐 or perhaps an additional `Array.isArray()` is ultimately much safer since technically a user should still be able to manually (mistakenly) set `translations` as an object since it is just a json column 👍

Reply:
Good one, the check for array would definitely be useful in preventing app crashes like above on invalid values. 👍